src/services/side-panels/models/item-wrapper.tsx

```diff
@@ -12,39 +11,46 @@ import { PanelComponentProps, SidePanelItem } from './side-panel-item';
 export class ItemWrapper {
   private static nextId = 0;
   public readonly id: number;
+  public iconId: number | undefined;
   public readonly item: SidePanelItem;
   private readonly player: KalturaPlayer;
+  private readonly upperBarManager: UpperBarManager;
   private panelItemComponentRef!: RefObject<PanelItemWrapper>;
   private removePanelComponentFn!: () => void;
-  private iconComponentRef: RefObject<IconWrapper> | undefined;
-  private removeIconComponentFn: (() => void) | undefined;
   private isActive: boolean;
   constructor(item: SidePanelItem, player: KalturaPlayer, onClick: (panelItemId: number) => void) {
     this.id = ++ItemWrapper.nextId;
     this.item = item;
     this.player = player;
+    this.upperBarManager = this.player.getService<UpperBarManager>('upperBarManager');
     this.isActive = false;
     this.injectPanelComponent();
-    if (item.iconComponent) this.injectIconComponent(onClick);
+    if (item.iconComponent) {
+      const itemId = this.id;
+      this.iconId = this.upperBarManager.addControl({
+        onClick: () => onClick(itemId),
+        component: this.item.iconComponent!
+      });
+    }
   }
 
   public activate(): void {
-    this.panelItemComponentRef.current!.on();
-    if (this.item.iconComponent) this.iconComponentRef!.current!.on();
+    this.panelItemComponentRef?.current?.on() || setImmediate(() => this.panelItemComponentRef!.current!.on());
```

Comment:
`setImmediate` has weak support of browsers: https://caniuse.com/?search=setImmediate
Might be `setTimeout` can be used here instead.